netbench/cdk/src/main/java/com/aws/PeeringStackProps.java

```diff
@@ -0,0 +1,111 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.aws;
+
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.Environment;
+import software.amazon.awscdk.services.ec2.Vpc;
+
+public interface PeeringStackProps extends StackProps {
```

Comment:
is there a need for this to be an interface? I'm a little confused of what this is for, it seems the `PeeringStackProps` extends `StackProps`, but the anonymous class below would just use all default values for StackProps since you haven't overriden any of the StackProps interface. Could it just implement StackProbs instead?

Reply:
Honestly, I was just following the example from this tutorial: https://cdkworkshop.com/50-java/40-hit-counter/100-api.html. Are you suggesting I change PeeringStackProps to a class that implements StackProps?

Reply:
In that example `HitCounterProps` doesn't extend `StackProps`. Is there a reason you need to extend the `StackProps` interface?

Reply:
Yeah `PeeringStackProps` is for the `PeeringStack` which extends `Stack`, so it needs to pass a `StackProps` object with the `super()` call, mainly to set the environment of the stack.

Reply:
It seems like nothing from `Stack` is able to be changed from the defaults though? Would it work the same if you called the `super` constructor that doesn't have the `StackProps` input like the `HitCounter.java` example?

Reply:
In my case I need to pass `Environment` in with the `super()` call as my stacks cannot be region-agnostic. I could technically create a separate `StackProps` object to pass in though, instead of extending it.

Reply:
I see now, `getEnv` is overriding the method in `StackProps` not `PeeringStackProps`, I didn't notice that. I think it should be ok then

Reply:
Right, sorry, I should have pointed that out earlier. Thank you for your feedback!